pkg/reconciler/managed/reconciler.go

```diff
@@ -306,6 +308,15 @@ type ExternalObservation struct {
 
 // An ExternalCreation is the result of the creation of an external resource.
 type ExternalCreation struct {
+	// ResourceLateInitialized should be true if the managed resource's spec was
+	// updated during its creation. A Crossplane provider may update a
+	// managed resource's spec fields or metadata in order to store the persistent
+	// ID of the external resource if it's generated by the Provider API.
+	// Note that changes to the spec will be persisted to api-server but status
+	// changes will be lost in this reconciliation if ResourceLateInitialized is
+	// true.
+	ResourceLateInitialized bool
```

Comment:
I wonder if it would be worth defining a new concept that is separate from "Late Initialization" for this case that is specific to identification information. However, it does fall into the category of late initialization, so maybe we should just be consistent

Reply:
I had the same thought. Intuitively I initially considered this to be different from late initialisation, though I can see the case for it being considered another form of late initialisation. Are we aware of cases in which the create call "late inits" a managed resource in any way other than setting its external name?

Reply:
> Intuitively I initially considered this to be different from late initialisation, though I can see the case for it being considered another form of late initialisation.

I feel like it falls under late-init. There is a default assigned by the API and we fetch it to update the not-given user input. One difference from the late-init we do in `Observe` is that, we override the user's input even if they did assign a value. Though if we go with the nuke option that [strictly updates only the external-name annotation](https://github.com/crossplane/crossplane-runtime/pull/228#discussion_r524164123), calling this late-init would be misleading.

> Are we aware of cases in which the create call "late inits" a managed resource in any way other than setting its external name?

I'm not sure but it might be possible where there are more than one string used to identify the resource and both could be generated by the API.

Reply:
> One difference from the late-init we do in Observe is that, we override the user's input even if they did assign a value.

I'll let you make the call here, but this doesn't feel like a late init to me.